src/ExpertExtender/ExpertExtender.CalendarHint.js

```diff
@@ -94,15 +94,17 @@
 				return;
 			}
 
-			// Are we in the interesting range
-			if( !( value.year() > 1581 && value.year() < 1930 && value.precision() > 10 ) ) {
+			var assumeCalendar = value.getYear() <= 1581 || value.getYear() >= 1930;
+
+			if( assumeCalendar || value.getOption( 'precision' ) <= 10 ) {
 				this.$calendarhint.hide();
 				return;
 			}
 
-			var msg = this._messageProvider.getMessage(
-				this._prefix + '-' + value.calendar().toLowerCase()
-			);
+			var calendarModel = value.getOption( 'calendarModel' ),
```

Comment:
So, you did not test the pull request after your additional changes in DataValuesJavaScript? What do you expect `value.getOption( 'calendarModel' )` to return being aware that `value` is a `TimeValue` object (https://github.com/wmde/DataValuesJavascript/blob/master/src/values/TimeValue.js)? If it is nor the URI, something is wrong as the call to `TimeValue.getCalendarModelKeyByUri` is just two lines below.